### DISCUSSION TITLE
trying curl for embedded

### DISCUSSION THREAD

**Discussion by abhishekzimetrics** [curl/curl#12143] [General]
Hi,
I want to implement the curl library on the stm32 board, is this possible??
if yes, then anyone can help me with this... While trying to build a static library in stm32 IDE I've attached the lib and include in my stm32 generated code and after all the setups when building the project, I got the including error as sys/socket.h: No such file.
Would anyone be able to help me with this??

Thanks & Regards,
Abhishek D

**Comment by dfandrich** [comment]
The STM32 is an ARM CPU, and an ARM CPU is powerful enough to run curl. But,
these are microcontrollers that have as little as a few KB of flash and RAM,
while curl needs more than 100 KB of flash and probably considerably more than
that of RAM. And that's not including the C support libraries, TCP/IP stack,
network and filesystem drivers, RTOS, etc. It looks like even the high-end
STM32 MCUs have only 1 MB flash and just 128 KB or so of RAM. That's really not
enough to try to shoehorn all you'd need to squeeze in a reasonable curl
installation.

**Reply by slugvision** [reply]
Hi

Along the same line - will curl work on a Raspberry Pi Pico, which has 2 MB flash and 256 k RAM? 
Controller is ARM Cortex-M0+ Compiler is GCC ... arm-none-eabi

**Reply by bagder** [reply]
If you get a TCP stack in there, and you opt to use one of the smallest TLS implementations then it seems doable. But then you don't have anywhere to store the content that you download...

**Reply by slugvision** [reply]
Hi Badger

Thanks, I will give it a try as I only need to download a few kilobytes of data.

Btw, would Tiny Curl be a better choice ? How much flash/RAM does that package need?

Best regards, Ole
